src/sdk/index.ts

```diff
@@ -419,36 +343,97 @@ export async function createPlayground(
  * @return {string} - The URL of the playground (as a string).
  */
 export function getPlaygroundUrl(options: EmbedOptions = {}): string {
-  const { appUrl, params, config, import: x, ...otherOptions } = options;
-  const configParam =
-    typeof config === 'string'
-      ? { config }
-      : config && typeof config === 'object' && Object.keys(config).length
-        ? { x: 'code/' + compressToEncodedURIComponent(JSON.stringify(config)) }
-        : {};
-
-  let encodedParams;
+  const {
+    appUrl = 'https://livecodes.io',
+    params = {},
+    config = {},
+    headless,
+    import: importFrom,
+    lite,
+    loading = 'lazy',
+    template,
+    view,
+  } = options;
+  const isHeadless = options.view === 'headless' || headless; // for backwards compatibility;
+
+  const playgroundUrl = new URL(appUrl);
+  const searchParams = new URLSearchParams();
+
+  if (lite) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `Deprecation notice: "lite" option is deprecated. Use "config: { mode: 'lite' }" instead.`,
+    );
+    if (typeof config === 'object' && config.mode == null) {
+      config.mode = 'lite';
+    } else {
+      playgroundUrl.searchParams.set('lite', 'true');
+    }
+  }
+  if (view) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `Deprecation notice: The "view" option has been moved to "config.view". For headless mode use "headless: true".`,
+    );
+    if (typeof config === 'object' && config.view == null && view !== 'headless') {
+      config.view = view;
+    } else {
+      playgroundUrl.searchParams.set('view', view);
+    }
+  } else if (typeof config === 'object') {
+    if (Object.keys(config).length > 0) {
+      playgroundUrl.searchParams.set('config', 'sdk');
+    }
+  } else {
+    throw new Error(`"config" is not a valid URL or configuration object.`);
+  }
+
+  if (typeof config === 'string') {
+    try {
+      new URL(config);
+      playgroundUrl.searchParams.set('config', encodeURIComponent(config));
+    } catch {
+      throw new Error(`"config" is not a valid URL or configuration object.`);
+    }
+  } else if (config && typeof config === 'object' && Object.keys(config).length > 0) {
+    playgroundUrl.searchParams.set('config', 'sdk'); // fixme: not sure of this one
+    if (config.title && config.title !== 'Untitled Project') {
+      playgroundUrl.searchParams.set('title', config.title);
+    }
+    if (config.description && config.description.length > 0) {
+      playgroundUrl.searchParams.set('description', config.description);
+    }
+    const hashParams = new URLSearchParams();
+    hashParams.set('x', 'code/' + compressToEncodedURIComponent(JSON.stringify(config)));
+    playgroundUrl.hash = hashParams.toString();
+  }
+
+  // handle params
   if (params && typeof params === 'object') {
```

Comment:
`params` already has a default value of `{}`.
let's add another condition here `&& Object.keys(params).length > 0`